sdk/orm-django/django_encryption/fields.py

```diff
@@ -42,8 +44,163 @@ class _RaiseError:
+
+    objects = EncryptedBatchManager()
+
+    class Meta:
+        abstract = True
+
+
+# EncryptedMixinDescriptor is a descriptor wrapping access to fields inheriting from EncryptedMixin
+# it allows us to:
+#   lazily decrypt the field value when it is accessed, and then caching the result
+#   bulk decrypt values
+class EncryptedMixinDescriptor(DeferredAttribute):
+
+    def __init__(self, field):
+        super().__init__(field)
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            return DeferredAttribute.__get__(self, instance, owner)
+        if hasattr(instance, 'decrypted_' + self.field.name):
+            decrypted_value = getattr(instance, 'decrypted_' + self.field.name)
+            return decrypted_value
+        if hasattr(instance, 'encrypted_' + self.field.name):
+            encrypted_value = getattr(instance, 'encrypted_' + self.field.name)
+            if encrypted_value is None:
+                setattr(instance, 'decrypted_' + self.field.name, None)
+                return None
+
+            transformation = None
+            if hasattr(instance, '_transform_fields'):
+                transformation = instance._transform_fields.get(
+                    self.field.name)
+            decrypted_value = self.field.get_decrypted_value(
+                encrypted_value, transformation=transformation)
+            setattr(instance, 'decrypted_' + self.field.name, decrypted_value)
+            return decrypted_value
+        return DeferredAttribute.__get__(self, instance, owner)
+
+    def __set__(self, instance, value):
+        if instance is None:
+            raise AttributeError("Can only be set on instances")
+        if value is None:
+            setattr(instance, 'decrypted_' + self.field.name, None)
+            setattr(instance, 'encrypted_' + self.field.name, None)
+            return
+        if isinstance(value, tuple):
+            # we got an encrypted value
+            value = value[1]
+            setattr(instance, 'encrypted_' + self.field.name, value)
+            return
+        # we got a decrypted value
+        setattr(instance, 'decrypted_' + self.field.name, value)
+
+    # get_prefetch_queryset is called by django when prefetching related objects
+    # this function allows django's queries to believe that the encrypted fields
+    # are like foreign keys and so can be prefetched
+    def get_prefetch_queryset(self, instances, queryset=None):
+        encrypted_attr_name = 'encrypted_' + self.field.name
+        encrypted_values = [getattr(instance, encrypted_attr_name)
+                            for instance in instances]
+        transformation = None
+        # we could actually have a separate transformation for each instance, but we are assuming
+        # that it's the same transformation for all of them
+        if hasattr(instances[0], '_transform_fields') and self.field.name in instances[0]._transform_fields:
```

Comment:
Can you export the inline condition to a func which describe better what you check ?

Reply:
That wouldn't really make sense IMO. I'm only checking if both _transform_fields is defined and that it contains a value. I only need to do this because _transform_fields is not necessarily defined. Extracting this outside won't make it more readable.